swap_meet/vendor.py

```diff
@@ -1,2 +1,64 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        """gets items in inventory of given category"""
+        items_by_category = []
+        
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+
+        return items_by_category
+    
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
```

Comment:
Nice use of a guard clause here to ensure the items we need are in the inventories. Notice that you _could_ have done the same error handling strategy as you employed earlier (`.index()` also raises `ValueError` if the item isn't found).